tests/core/test_summary.py

```diff
@@ -23,10 +23,10 @@ def gcd_with_metrics(gcd_chip):
             for metric in gcd_chip.getkeys('flowgraph', flow, step, index, 'weight'):
                 gcd_chip.set('flowgraph', flow, step, index, 'status', NodeStatus.SUCCESS)
                 gcd_chip.set('metric', metric, str(dummy_data), step=step, index=index)
-                prev_step = steps.index(step) - 1
+                prev_step = steps.index((step, index)) - 1
```

Comment:
this test should be corrected (it looks to have been wrong before too), just picking the previous does not guarantee its the previous node.